### ISSUE TITLE
Support variable width row headers

### ISSUE THREAD

**Issue by chahu** [handsontable/handsontable#475]
I'm working on a project which is attempting to use handsontable, however we're running up against the fact that the row-header-column is a fixed width (and not very wide).  This cause the vast majority of our row header labels to be truncated.  One alternative is to tweak the CSS and widen that first column a fixed amount, but this is not an option as the data we are displaying could vary in length/width.  I've also tried changing the CSS and commenting out some of the lines with "width: 0" -- this "works" in that the row-header-column does indeed become variable width, but the scrolling logic becomes buggy.

Long story short, how can I get the below to show my row headers (without having a fixed-width column) and keep the nice scrolling functionality:

[http://jsfiddle.net/4587J/6/](http://jsfiddle.net/4587J/6/)

Thanks!
Charlie

**Comment by tartor** [comment]
Workaround 
in the config implement afterLoadData callback to change the width of all the elements that have class name 'col.rowHeader'

var ht = new Handsontable(
                document.getElementById(id),
                {
                    afterLoadData: function(){
                        var rowHeaders = document.getElementById(id).querySelectorAll('col.rowHeader');
                        if (rowHeaders.length != 0) {
                            for (var j = 0; j < rowHeaders.length; j++) {
                                rowHeaders[j].style.width = "200px";
                            }
                        }
                    }
                }
            );

where id is the id of the html element which holds the handsontable

**Comment by realtebo** [comment]
Is there any CSS way to defined a 'auto-width' for row headers ?

**Comment by donmccurdy** [comment]
Could this issue be reopened? The JSFiddle posted by the OP ([which I've updated with fixed links to CDN js/css](http://jsfiddle.net/donmccurdy/vzp31798/)) is still showing row headers fixed at 50px. As pointed out, the 50px can be overridden in CSS, but this isn't ideal.

@realtebo No, not without using JavaScript to calculate what the magic width should be, and then applying it in CSS.

**Comment by AMBudnik** [comment]
@donmccurdy this issue is continued (and marked as a bug) at https://github.com/handsontable/handsontable/issues/2868
